frontend/src/metabase/query_builder/components/AggregationPopover.jsx

```diff
@@ -250,14 +250,15 @@ export default class AggregationPopover extends Component {
 
     // we only want to consider active metrics, with the ONE exception that if the currently selected aggregation is a
     // retired metric then we include it in the list to maintain continuity
-    const metrics =
-      showMetrics && tableMetadata.metrics
-        ? tableMetadata.metrics.filter(
-            metric =>
-              !metric.archived ||
-              (selectedAggregation && selectedAggregation.id === metric.id),
-          )
-        : [];
+    const metrics = tableMetadata.metrics
+      ? tableMetadata.metrics.filter(metric =>
+          showMetrics
+            ? !metric.archived ||
+              (selectedAggregation && selectedAggregation.id === metric.id)
+            : // GA metrics are more like columns, so they should be displayed even when showMetrics is false
+              metric.googleAnalyics,
+        )
+      : [];
```

Comment:
@tlrobinson in lieu of a larger review, what do you think about putting this here? Previously we were creating a `new Metadata` that filtered out non-Google Analytics metrics. That felt ugly and I figured the PartialQueryBuilder should use the full Metadata and filter it when displaying. Is there a better spot for this logic?

Definitely feel free to comment on whatever else here if you want, but there's a big diff and most of this PR is moving things around and small refactors.

Reply:
Yeah I think this is better than a custom Metadata